beacon-chain/blockchain/chain_info.go

```diff
@@ -329,24 +341,25 @@ func (s *Service) IsOptimisticForRoot(ctx context.Context, root [32]byte) (bool,
 	if err != protoarray.ErrUnknownNodeRoot && err != doublylinkedtree.ErrNilNode {
 		return false, err
 	}
+	// if the requested root is the headroot we should treat the
+	// node as optimistic. This can happen if we pruned INVALID
+	// nodes and no viable head is available.
+	headRoot, err := s.HeadRoot(ctx)
+	if err != nil {
+		return true, err
+	}
+	if bytes.Equal(headRoot, root[:]) {
+		return true, nil
+	}
```

Comment:
This is confusing. I think often readers will miss the fact that we only end up here if the root is not in fork choice store

Reply:
I'll update the comment